packages/react-components/priority-overflow/src/createResizeObserver.ts

```diff
@@ -15,6 +15,7 @@ export function observeResize(elementToObserve: HTMLElement, callback: ResizeObs
     return () => null;
   }
 
+  // eslint-disable-next-line no-restricted-globals
```

Comment:
we could probably do following to avoid turning off the lint ?

`let resizeObserver: typeof GlobalResizeObsever | undefined = new GlobalResizeObsever(callback);`

on second though wondering why we need to explicitly type this `let` statement...

why do we need to set it to `undefined` ? I'd expect that disconnect does what it is supposed to do = not leaving hanging references and properly garbage collected . cc @ling1726 

cant we do just this ?

```ts
const resizeObserver = new GlobalResizeObsever(callback);
resizeObserver.observe(elementToObserve);

return () => {
    resizeObserver.disconnect();
};
```

Reply:
Changing to `const` seems fine to me but I defer to @ling1726 as the original author who may be intentionally doing this (to address a GC issue perhaps?).

Reply:
> let resizeObserver: typeof GlobalResizeObsever | undefined = new GlobalResizeObsever(callback);

<img width="1074" alt="Code_yKo3jOBXSz" src="https://github.com/microsoft/fluentui/assets/93940821/1ad795b4-74e1-47d9-8589-5ca99aeb136c">

Any ideas here?

Reply:
Also realized we have a typo :)

> GlobalResizeObsever --> GlobalResizeObserver

Reply:
so with the type fix , `typeof GlobalResizeObsever` pattern wont work ?

Reply:
update: typeof GRO will return `Constructor` type which is not the instance type we wanna use, thus we need to do `InstanceType<typeof GlobalResizeObserver> | undefined`

Reply:
That did the trick! Thanks!